lsp/server/src/server.ts

```diff
@@ -323,6 +394,23 @@ connection.onDidChangeWatchedFiles(_change => {
 	connection.console.log('We received an file change event');
 });
 
+connection.onShutdown(() => {
+	connection.workspace.getWorkspaceFolders().then((folders: null|WorkspaceFolder[]|void) => {
+		if (folders) {
+			folders.forEach((folder) => {
+				const path = getPath(folder);
+				execSync(`docker container stop sa4u_z3_server_${path}`);
```

Comment:
We need to wrap `execSync` in a try/catch block, in case the container exited erroneously early.
Note: I encountered this issue when validating this change.

Reply:
Added try/catch block.